packages/melody-streams/src/operators/withElement.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright 2019 trivago N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Subject } from 'rxjs';
+
+export const withElement = handler => {
+    const subj = new Subject();
+    const refHandler = el => handler(el).subscribe(next => subj.next(next));
+    return [refHandler, subj];
+};
```

Comment:
It could be handy to accept multiple handlers, e.g. when you want to have a click stream and attach an intersection observer to the same element:

```js
const createClickStream = el => fromEvent(el, 'click');
const createVisibilityStream = el => { /* ... */ };

const Component = ({ props, subscribe }) => {
    const [refHandler, clicks, isVisible] = withElement(
        createClickStream,
        createVisibilityStream
    );

    return combine(
        props,
        isVisible,
        clicks.pipe(
            tap(() => console.log('clicked')),
            ignoreElements()
        )
    );
};
```
Implementation
```js
export const withElement = (...handlers) => {
    const subjects = handlers.map(() => new Subject());
    const refHandler = el => {
        const observables = handlers.map((handler, i) =>
            handler(el).map(next => [i, next])
        );
        return merge(...observables).subscribe(([i, next]) =>
            subjects[i].next(next)
        );
    };

    return [refHandler, ...subjects];
};
```